OPAL/br/src/main/scala/org/opalj/br/fpcf/properties/Alias.scala

```diff
@@ -22,15 +22,53 @@ sealed trait AliasPropertyMetaInformation extends PropertyMetaInformation {
  * An alias property provides information about the relationship of the memory locations of the associated entities.
  *
  *  - [[NoAlias]] can be assigned to a pair of entities, iff no execution path exists where both entities refer to the same memory location.
+ *   This implies that there is a definite separation of the memory locations of the two entities, and no interference can occur when one of the entities is referenced.
+ *   Note that this does not imply that, for example, the fields of the entities cannot refer to the same memory location.
+ *   Example:
+ *   {{{
+ *     Object m(Object a) {
+ *       Object b = new Object();
+ *       return b;
+ *     }
+ *   }}}
+ *   In this example, the formal parameter `a` and the local variable `b` are guaranteed to never refer to the same memory location,
+ *   as `b` is always assigned to a new object that the formal parameter can't possibly be pointing to since the object has not been created before the method call.
+ *   This ensures that the method's return value and the formal parameter never point to the same memory location.
```

Comment:
This is a very interesting case actually. I agree that during the method call, they can never alias and that's probably what a user would be interested in. But typically, these values will be abstractly combined over all (context-insensitive) or at least some (context-sensitive) calls, which I guess by definition could make them alias if the return value of one invocation can be passed as the parameter of another invocation of that method. What does the literature say about this?

Reply:
Typically, aliasing is defined via the overlapping of the memory locations. Even if you look at it with all calls being combined, it still shouldn't be an alias, because they still point to distinct memory locations, despite being created at the same position (in code). You could of course view aliasing as a "created at the same location in code (same new instruction)" but I don't see how that would be helpful, because it doesn't really guarantee any useful properties.

But I think we should maybe use a more straight-forward example without parameters here to avoid any unnecessary confusion.